src/utils/waypoint.tsx

```diff
@@ -0,0 +1,53 @@
+import React from "react"; // { useState , FormEventHandler },
+import { useSelector, useDispatch } from "react-redux";
+import moment from "moment";
+
+import type { Waypoint } from "@bindings/protobufs/Waypoint";
+
+import {
+  selectActiveWaypoint,
+  selectIsWaypointEdit,
+} from "@features/device/deviceSelectors";
+import { requestNewWaypoint } from "@features/device/deviceActions";
+import { deviceSliceActions } from "@features/device/deviceSlice";
+
+// This file contains utils for: Delete waypoint and Toggle editWaypoint
+
+// Delete waypoint
+const useDeleteWaypoint = () => {
+  const dispatch = useDispatch();
+  const activeWaypoint = useSelector(selectActiveWaypoint());
+
+  return () => {
+    console.log("Deleted waypoint");
+
+    // Set expiry to time one second ago
+    if (activeWaypoint) {
+      const updatedWaypoint: Waypoint = {
+        ...activeWaypoint,
+        expire: Math.round(moment().valueOf() / 1000) - 1,
+      };
+
+      // Update waypoint and clear ActiveWaypoint
+      dispatch(requestNewWaypoint({ waypoint: updatedWaypoint, channel: 0 }));
+      dispatch(deviceSliceActions.setActiveWaypoint(null));
+
+      // Error
+    } else {
+      console.log("Error: No active waypoint");
+    }
+  };
+};
+
+// Toggles state of editWaypoint
+const useToggleEditWaypoint = () => {
+  const dispatch = useDispatch();
+  const isWaypointEdit = useSelector(selectIsWaypointEdit());
+  return () => {
+    dispatch(deviceSliceActions.setWaypointEdit(!isWaypointEdit));
+  };
+};
+
+// TODO: const useSendWaypoint = (waypoint: Waypoint) => {}
```

Comment:
What is this for?

Reply:
I figured that if we were going to be sending waypoints from the `WaypointEditMenu,` `MapView` (for new waypoints), and eventually `WaypointConfig`, it might be worth having a common one that they share instead of having a new function every time

Reply:
Sounds good!